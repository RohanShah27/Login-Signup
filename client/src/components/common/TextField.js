// Functional text field component
import React, { useState } from "react";
import "../../assets/styles/text-field.css";
export default function TextField({
  id,
  value,
  label,
  type,
  handleChange,
  onBlur,
  onFocus,
  minChar,
  maxChar,
  error,
  placeholder,
}) {
  // Hook to declare the focus state of the field and the setter function
  const [focus, setFocus] = useState(false);
  return (
    <div className="main-container">
      <label className="label" htmlFor={id + "_element"}>
        {label}
      </label>
      <div
        // based on fields like disabled error and warning change the class names to present different
        //  styles to the user
        className={`text-field-container ${
          error ? " error-input" : focus ? "text-field-container-focus" : ""
        }`}
      >
        <input
          value={value}
          id={id + "input"}
          placeholder={placeholder}
          type={type ? type : "text"}
          className="input-text-field"
          onChange={handleChange}
          // Event to change border color when user has clicked on the field
          onFocus={onFocus ? onFocus : () => setFocus(true)}
          // Event to change border color when user has clicked outside the field field
          onBlur={onBlur ? onBlur : () => setFocus(false)}
          minLength={minChar} //minimum number of inputs
          maxLength={maxChar} //maximum number of inputs
        />
      </div>
      {/* Display the error message if any */}
      {error ? (
        <span
          className="input-error-message"
          id={"error-message" + id}
          key={error}
        >
          {error}
        </span>
      ) : null}
    </div>
  );
}
