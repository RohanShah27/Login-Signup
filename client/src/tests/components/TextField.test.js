import React from "react";
import { shallow } from "enzyme";
import TextField from "../../components/common/TextField";

const id = "test";
const value = "value";
const label = "test label";
const type = "password";
const handleChange = jest.fn();
const onBlur = jest.fn();
const onFocus = jest.fn();
const minChar = 5;
const maxChar = 10;
const error = "";
const placeholder = "password";
const wrapper = shallow(
  <TextField
    id={id}
    value={value}
    label={label}
    type={type}
    handleChange={handleChange}
    onBlur={onBlur}
    onFocus={onFocus}
    minChar={minChar}
    maxChar={maxChar}
    error={error}
    placeholder={placeholder}
  />
);

describe("Test suite for text field fnc", () => {
  it("should render the component", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("simulate on change and on blur", () => {
    let event = {
      target: { value: "abc" },
    };
    wrapper.find("#testinput").simulate("change", event);
    expect(handleChange).toBeCalled();
    wrapper.find("#testinput").simulate("blur");
    expect(onBlur).toBeCalled();
  });
  it("should check for props", () => {
    expect(wrapper.find("#testinput").props().maxLength).toBe(maxChar);
    expect(wrapper.find("#testinput").props().minLength).toBe(minChar);
    expect(wrapper.find("#testinput").props().placeholder).toBe(placeholder);
    expect(wrapper.find("#testinput").props().type).toBe(type);
    expect(wrapper.find("#testinput").props().value).toBe(value);
  });
});
