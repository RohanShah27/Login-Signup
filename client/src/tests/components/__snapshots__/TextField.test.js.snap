// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for text field fnc should render the component 1`] = `ShallowWrapper {}`;
